charts/k8s-watcher/templates/clusterrole.yaml

```diff
@@ -423,6 +423,23 @@ rules:
       - get
       - watch
 {{- end}}
+
+{{- if .Values.watcher.resources.allowReadAll }}
+  - apiGroups: ["*"]
+    resources: ["*"]
+    verbs:     ["get", "watch", "list"]
+{{- end}}
+
+{{- $numOfCustomApiGroups := len .Values.watcher.resources.customAPIGroups }}
+{{- $shouldLimitApiGroups := gt $numOfCustomApiGroups 0 }}
+{{- if and $shouldLimitApiGroups (not .Values.watcher.resources.allowReadAll) }}
+  {{- range $cag := .Values.watcher.resources.customAPIGroups }}
```

Comment:
Do we really need this complexity? 
What will happen if the customAPIGroups will be defined although the allowReadAll is true?

Reply:
I agree that we can just iterate over the customAPI groups regardless of the read all flag
we also have this duplication in all other native resources